docs/populators.md

```diff
@@ -0,0 +1,171 @@
+from STACpopulator.extensions.thredds import THREDDSCatalogDataModel
+
+
+class RDPSDataModel(THREDDSCatalogDataModel):
+    """Data model for RDPS NetCDF datasets."""
+
+    cf: CFHelper
+    file: FileHelper
+```
+
+## 4. Populator
+
+Populators are defined in a specific directory created under `STACpopulator/implementations/`, to which we will refer in the following as the populator's directory. The name of this newly created directory must follow the naming convention: `STACpopulator/implementations/IMPLEMENTATION_AUTHOR/`. For instance, `STACpopulator/implementations/RDPS_CRIM/` for the RDPS populator.
+
+### 4.1. Creating the populator class
+
+A populator class inherits from the [`STACpopulatorBase`](../STACpopulator/populator_base.py#L26) abstract class and represents the final data layer in a `stac-populator` implementation. It specifies the corresponding `DataModel` type and implements the abstract method `create_stac_item(...)`, which creates a data model instance for each STAC item while applying the relevant extensions described earlier. A populator class also inherits the `def ingest(...)` method, which is called when the command associated with the populator implementation is executed, triggering the data ingestion process.
+
+The example below shows the [`RDPSpopulator`](../STACpopulator/implementations/RDPS_CRIM/add_RDPS.py#L14) class. 
+
+
+```python
+from typing import Any
+
+from STACpopulator.extensions.rdps import RDPSDataModel
+from STACpopulator.populator_base import STACpopulatorBase
+
+
+class RDPSpopulator(STACpopulatorBase):
+    """Populator that creates STAC objects representing RDPS data from a THREDDS catalog."""
+
+    data_model = RDPSDataModel
+    item_geometry_model = None  # Unnecessary, but kept for consistency
+
+    def create_stac_item(self, item_name: str, item_data: dict[str, Any]) -> dict[str, Any]:
+        """Return a STAC item."""
+        model = self.data_model.from_data(item_data, session=self._session)
+        return model.stac_item()
+```
+
+### 4.2. Adding populator CLI command
+
+To be able to invoke the populator from the CLI, following updates must be performed.
+
+1. Copy the `def add_parser_args(...)` and `def runner(...)` methods from the [`STACpopulator/implementations/RDPS_CRIM/add_RDPS.py`](../STACpopulator/implementations/RDPS_CRIM/add_RDPS.py) file.
+2. Update the Python docstrings and CLI help strings to reflect the new populator’s name and specific behavior.
+3. Export these methods in the `__init__.py` file of the populator's directory. 
+
+For instance, in `STACpopulator/implementations/RDPS_CRIM/__init__.py` for the RDPS populator.
+
+```python
+from .add_RDPS import add_parser_args, runner
+
+__all__ = ["add_parser_args", "runner"]
+```
+
+4. Register the new populator implementation in [`STACpopulator/implementations/__init__.py`](../STACpopulator/implementations/__init__.py)
+
+```python
+__all__ = [..., "RDPS_CRIM", "IMPLEMENTATION_AUTHOR"]
```

Comment:
Explicitly say somewhere "Replace IMPLEMENTATION_AUTHOR with the name of your new implementation's package in the implementations directory"